src/Compilers/Core/Portable/Compilation/IOperationVisitor.cs

```diff
@@ -0,0 +1,746 @@
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitFieldInitializer(IFieldInitializer operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitPropertyInitializer(IPropertyInitializer operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitArrayCreationExpression(IArrayCreationExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitArrayInitializer(IArrayInitializer operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitAssignmentExpression(IAssignmentExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitCompoundAssignmentExpression(ICompoundAssignmentExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitIncrementExpression(IIncrementExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitParenthesizedExpression(IParenthesizedExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitLateBoundMemberReferenceExpression(ILateBoundMemberReferenceExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitUnboundLambdaExpression(IExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitDefaultValueExpression(IExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitTypeParameterObjectCreationExpression(IExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitInvalidStatement(IStatement operation)
+        {
+            this.DefaultVisit(operation);
+        }
+
+        public virtual void VisitInvalidExpression(IExpression operation)
+        {
+            this.DefaultVisit(operation);
+        }
+    }
+
+    /// <summary>
+    /// Represents a <see cref="IOperation"/> visitor that visits only the single IOperation
+    /// passed into its Visit method with an additional argument of the type specified by the 
+    /// <typeparamref name="TArg"/> parameter and produces a value of the type specified by 
+    /// the <typeparamref name="TResult"/> parameter.
+    /// </summary>
+    /// <typeparam name="TArg">
+    /// The type of the additional argument passed to this visitor's Visit method.
+    /// </typeparam>
+    /// <typeparam name="TResult">
+    /// The type of the return value of this visitor's Visit method.
+    /// </typeparam>
+    public abstract class IOperationVisitor<TArg, TResult>
```

Comment:
I see. I was mostly thinking about using visitor from Roslyn analyzers' point of view. But in your case, your implementation of IOperation is separate from Roslyn's (wrapper of bound nodes) and independent, is that correct?